front/src/applications/operationalStudies/hooks/useScenarioData.ts

```diff
@@ -189,8 +205,8 @@ const useScenarioData = (scenario: ScenarioResponse, infra: InfraWithState) => {
       infra.state === 'CACHED' &&
       (timetableItemsWithDetails.length === 0 || pacedTrainWithDetails.length === 0)
     ) {
-      const timetableItemIds = timetableItems.map((timetableItem) => timetableItem.id);
-      setTimetableItemIdsToFetch(timetableItemIds);
+      const timetableItemIdsList = timetableItems.map((timetableItem) => timetableItem.id);
+      setTimetableItemIdsToFetch(timetableItemIdsList);
```

Comment:
Do we need to recompute it here ? Can't we just use `timetableItemIds` ? (if yes, maybe we should change the array of dependencies)
```suggestion
      setTimetableItemIdsToFetch(timetableItemIds);
```

Reply:
Hmm isn't it safer not to depend on a state that is fare away from where this function will be called ? And just depend on the function param ?

Reply:
Ok fine for now :+1: